runtime/auth/aws-credentials/common/src/aws/smithy/kotlin/runtime/auth/awscredentials/CachedCredentialsProvider.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.auth.awscredentials
```

Comment:
**Comment**: The package **awscredentials** doesn't feel quite right here. These are generic credentials utilities and it seems like just **credentials** would be more appropriate.

Reply:
Discussed offline. Keeping this as **awscredentials** makes more sense. `Credentials` and `CredentialsProvider` already live in this package and `Credentials` in particular is AWS-specific.